collections/max_of.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Applies the given selector to all elements of the given collection and returns the max value of all elements
+ *
+ * Example:
+ *
+ * ```ts
+ * import { maxOf } from "./max_of.ts"
+ * import { assertEquals } from "../testing/asserts.ts"
+ *
+ * const inventory = [
+ *      { name: "mustard", count: 2 },
+ *      { name: "soy", count: 4 },
+ *      { name: "tomato", count: 32 },
+ * ];
+ * const maxCount = maxOf(inventory, (i) => i.count);
+ *
+ * assertEquals(maxCount, 32);
+ * ```
+ */
+export function maxOf<T>(
+  array: readonly T[],
+  selector: (el: T) => number,
+): number;
```

Comment:
Is this return type correct? I think this can return undefined. So maybe we should type it `number | undefined`?

Reply:
Same applies for the `bigint` signature @Hakizu !

Missed that one, thanks for the catch @kt3k